router/internal/batch/batch.go

```diff
@@ -0,0 +1,107 @@
+package batch
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/go-chi/chi/v5"
+	"github.com/goccy/go-json"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"io"
+	"net/http"
+)
+
+const unlimitedBatchEntries = 0
+
+func Handler(maxEntriesPerBatch, maxRoutines int, handlerSent http.Handler, tracerProvider *sdktrace.TracerProvider) http.Handler {
+	tracer := tracerProvider.Tracer(
```

Comment:
It's good practice to not create a tracer per request, although I believe it is cached. Could you create this before once?

Reply:
This is only created once, and not per request, it is outside of the handler func (I will make it a bit more explicit).